spec/support/saml_response_doc.rb

```diff
@@ -138,6 +138,16 @@ def metadata
     metadata_nodeset[0]
   end
 
+  def metadata_name_id_format(name)
+    name_id_format_regex = /^urn:oasis:names:tc:SAML:\d\.\d:nameid-format:#{name}$/
+    metadata.xpath(
+      './ds:IDPSSODescriptor/ds:NameIDFormat',
+      ds: Saml::XML::Namespaces::METADATA,
+    ).
+    map { |f| f.children[0].content }.
+    find { |f| f.match(name_id_format_regex) }
```

Comment:
I would consider simplifying the regex to just a simple substring and maybe trying to leverage xpath `contains()` to do the hard work? (I haven't tried this but the gist should work

```
metadata.xpath(
  "./ds:IDPSSODescriptor/ds:NameIDFormat[contains(text(), '#{name})]",
).first.content
```

Reply:
Ah I already merged! I'll open a follow-up JIRA ticket and will take a look at refactoring.

Reply:
Resolving in #4765